// containers/layout/index.js
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { getGallery, setFilter } from 'actions';
import Header from 'components/Header';
import Filter from 'components/Filter';
import ItemList from 'components/ItemList';
import List from 'components/List';
import Button from 'components/Button';

class Layout extends Component {
	constructor(props) {
		super(props);
		this.interval = null;
	}
	componentDidMount() {
		this.props.getGallery();
	}
	handleAutoRefresh(autoRefreshOn) {
		if(autoRefreshOn) {
			clearInterval(this.interval)
		} else {
			this.interval = setInterval(() => {
				this.props.getGallery();
			}, 3000);
		}
	}
	render() {
		const { gallery, setFilter } = this.props;
		let galleryData = gallery.data && gallery.data.children ? gallery.data.children : [];
		console.log('gallery data', this.gallery);
		return (
		<div>
			<Header galleryList={this.gallery}/>
			<Button autoRefreshHandler={this.handleAutoRefresh.bind(this)}></Button>
			<Filter setFilter={setFilter}/>
			<List galleryData={galleryData}/>
		</div>
		)
	}
}
const mapStateToProps = (state) => {
	return {
		gallery: state.gallery
	}
};
const mapDispatchToProps = {
	getGallery,
	setFilter
};
export default connect(mapStateToProps, mapDispatchToProps)(Layout);
