// containers/layout/index.js
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { getGallery } from 'actions';
import Header from 'components/Header';
import Filter from 'components/Filter';
import ItemList from 'components/ItemList';
import List from 'components/List';
import Button from 'components/Button';

class Layout extends Component {
	constructor(props) {
		super(props);
		this.interval = null;
	}
	componentDidMount() {
		this.props.getGallery();
	}
	handleAutoRefresh(isInterval) {
		if(!isInterval) {
			clearInterval(this.interval)
		} else {
			this.interval = setInterval(() => {
				console.log('autorefresh')
			}, 1000);
		}
	}
	render() {
		this.gallery = this.props.gallery;
		console.log('gallery data', this.gallery);
		return (
		<div>
			<Header galleryList={this.gallery}/>
			<Button autoRefreshHandler={this.handleAutoRefresh.bind(this)}></Button>
		</div>
		)
	}
}
const mapStateToProps = (state) => {
	return {
		gallery: state.gallery
	}
};
const mapDispatchToProps = {
	getGallery
};
export default connect(mapStateToProps, mapDispatchToProps)(Layout);
